third-party/timelib/0001-add-CMakeLists.txt.patch

```diff
@@ -0,0 +1,41 @@
+From a8dbfdf74f99008b528a39b8674a8402613af003 Mon Sep 17 00:00:00 2001
+From: Scott Sandler <ssandler@slack-corp.com>
+Date: Mon, 16 Aug 2021 14:02:06 -0700
+Subject: [PATCH] add CMakeLists.txt to timelib
+
+---
+ CMakeLists.txt | 23 +++++++++++++++++++++++
+ 1 file changed, 23 insertions(+)
+ create mode 100644 CMakeLists.txt
+
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+new file mode 100644
+index 0000000..51df16f
+--- /dev/null
++++ b/CMakeLists.txt
+@@ -0,0 +1,23 @@
++cmake_minimum_required(VERSION 2.6)
++project(timelib C)
++message("Building timelib in ${CMAKE_CURRENT_SOURCE_DIR}")
++
++# adapted from timelibs FLAGS but without usan, asan, and with optimizations
++set(CMAKE_C_FLAGS "-O2 -ggdb3 -Wall -Werror -Wextra -Wmaybe-uninitialized -Wmissing-field-initializers -Wshadow -Wno-unused-parameter -Wno-implicit-fallthrough -DHAVE_STDINT_H -DHAVE_GETTIMEOFDAY -DHAVE_UNISTD_H -DHAVE_DIRENT_H -I.")
++
++# timelib requires re2c on two of its source files
++execute_process(
```

Comment:
not sure if this is the best way of doing this but timelib now has two files that needed `re2c`. this also adds `re2c` as a new build dep for hhvm.